packages/react-jsx-highcharts/src/index.js.flow

```diff
@@ -0,0 +1,182 @@
+// @flow
+import React from 'react';
+import type { Node, ComponentType, ElementConfig } from 'react';
+import type HighchartsType from 'highcharts';
+
+
+type HighchartsChartProps = {
+  children?: Node
+}
+
+export function HighchartsChart(props: HighchartsChartProps) { }
+
+type HighchartsSparklineProps = {
+  height?: number,
+  width?: number,
+  margin?: Array<number>,
+  style?: {},
+  plotOptions?: {},
+  series?: Node,// note series or children required
+} & HighchartsChartProps
+
+export class HighchartsSparkline extends React.Component<HighchartsSparklineProps> { }
+
+
+type Highcharts3dChartProps = {
+  alpha?: number,
+  axisLabelPosition?: string,
+  beta?: number,
+  depth?: number,
+  fitToPlot?: boolean,
+  frame?: {},
+  viewDistance?: number,
+} & HighchartsChartProps;
+
+export class Highcharts3dChart extends React.Component<Highcharts3dChartProps> {
+
+}
+
+type ChartProps = {
+  type?: string,
+  width?: number,
+  height?: number,
+  onAddSeries?: Function,
+  onAfterPrint?: Function,
+  onBeforePrint?: Function,
+  onClick?: Function,
+  onLoad?: Function,
+  onRedraw?: Function,
+  onRender?: Function,
+  onSelection?: Function,
+}
+export function Chart (props: ChartProps) {}
+type CreditsProps = {
+  enabled?: boolean
+}
+export function Credits (props:CreditsProps) { }
+type LoadingProps = {
+  isLoading?: boolean,
+  hideDuration?: number,
+  labelStyle?: {},
+  showDuration?: number,
+  style?: {},
+}
+export function Loading (props: LoadingProps) { }
+
+type LegendProps = {
+  children?: Node,
+  enabled?: boolean
+}
+type LegendTitleProps = {
+  children?: Node
+}
+function LegendTitle (props: LegendTitleProps) { }
+export function Legend (props: LegendProps) {
+
+}
+Legend.Title = LegendTitle;
+
+type PlotBandProps = {
+  id?: string,
```

Comment:
Since v3 `id` can also be a function (returning a string), so I guess we could

```js
type Id = string | () => string

type PlotBandProps = {
  id?: Id,
  // etc
}

type Axis = {
  id?: Id,
  // etc
}
```

Reply:
I was considering the id function to be internal implementation detail, but I can add it to typings also.